apps/molgenis-components/src/stores/sessionStore.ts

```diff
@@ -0,0 +1,119 @@
+import { defineStore } from "pinia";
+import { Ref, ref } from "vue";
+import { useCookies } from "vue3-cookies";
+import { request } from "../client/client";
+import {
+  IErrorMessage,
+  IResponse,
+  ISession,
+} from "../components/account/Interfaces";
+import { ISetting } from "../Interfaces/ISetting";
+
+const { cookies } = useCookies();
+const defaultSession = {
+  locale: "en",
+  settings: {} as Record<string, string>,
+};
+let graphqlError = "";
+
+export const useSessionStore = defineStore("session", () => {
+  let session: Ref<ISession> = ref(defaultSession);
+  updateSession();
+
+  function updateSession() {
+    getSession().then((newSession) => {
+      session.value = newSession;
+    });
+  }
+
+  function signin() {
+    updateSession();
+  }
+
+  async function signout() {
+    const data = await request("graphql", `mutation{signout{status}}`).catch(
+      (error: string) => (graphqlError = "internal server error" + error)
+    );
+    if (data.signout.status === "SUCCESS") {
```

Comment:
since you use typescript, can you create an enum for all these status magic strings?

Reply:
That might be possible, but for this PR we didn't really look in to it, since the relevant code has barely been changed, only moved.